packages/editor/src/components/block-styles/index.js

```diff
@@ -55,22 +52,16 @@ export function getActiveStyle( styles, className ) {
  * @return {string} The updated className.
  */
 export function replaceActiveStyle( className, activeStyle, newStyle ) {
-	let added = false;
-	let updatedClassName = compact( className
-		.split( ' ' )
-		.map( ( current ) => {
-			const trimmed = current.trim();
-			if ( activeStyle && trimmed === 'is-style-' + activeStyle.name ) {
-				added = true;
-				return newStyle.isDefault ? null : 'is-style-' + newStyle.name;
-			}
-			return trimmed;
-		} ) ).join( ' ' );
-	if ( ! added && ! newStyle.isDefault ) {
-		updatedClassName = updatedClassName + ' is-style-' + newStyle.name;
+	const newStyleName = 'is-style-' + newStyle.name;
+
+	const list = new TokenList( className );
+	if ( activeStyle ) {
+		list.replace( 'is-style-' + activeStyle.name, newStyleName );
```

Comment:
It should be noted that the behavior of `DOMTokenList#replace` is such that, if the original token is not found, then the new token is not added. Meaning we _assume_ here that `activeStyle` is present, otherwise the new style name will not be included.

Reply:
In fact, this behavior is to blame for the regression observed through the failing end-to-end test, resolved (with new unit test case) in 6fcf8a5.